xarray/core/ops.py

```diff
@@ -17,6 +18,10 @@
 
 try:
     import bottleneck as bn
+    if StrictVersion(bn.__version__) < StrictVersion('1.0'):
+        warnings.warn('xarray requires bottleneck version of 1.0 or greater.'
```

Comment:
fair enough. The logic gets spread out a bit more but we now only raise the warning when `rolling.__init__()` is called.